libraries/redcore/api/hal/hal.php

```diff
@@ -1376,16 +1388,19 @@ public function checkRequiredFields($data, $configuration)
 				{
 					if (is_null($data[(string) $field['name']]) || $data[(string) $field['name']] == '')
 					{
-						JFactory::getApplication()->enqueueMessage(
-							JText::sprintf('LIB_REDCORE_API_HAL_WEBSERVICE_ERROR_REQUIRED_FIELD', (string) $field['name']), 'error'
-						);
-
-						return false;
+						$errors[] = JText::sprintf('LIB_REDCORE_API_HAL_WEBSERVICE_ERROR_REQUIRED_FIELD', (string) $field['name']);
 					}
 				}
 			}
 		}
 
+		if (!empty($errors))
+		{
+			$this->apiErrors = array_merge($this->apiErrors, $errors);
```

Comment:
ok, can you do it for all 400 and above errors then?
Also, why did you kick this out? https://github.com/redCOMPONENT-COM/redCORE/pull/517/files#diff-e80c433a2a4be9b271e294c8f12c2605L1379
It will be removed from the response message container...